cli/spec.py

```diff
@@ -108,6 +110,27 @@ def resolve_value(self, arg_name, defaults=None):
         if self.value is None:
             self.value = defaults.get(self.arg_name)
 
+    def __eq__(self, other):
+        """
+        Checks if other value is equal to the current value.
+        """
+        if isinstance(other, ValueArgument):
+            return self.value == other.value
+        else:
+            return self.value == other
+
+    def __lt__(self, other):
```

Comment:
self.value for YamlFileArgument is the dict. So this will compare two dict in case for YamlFileArgument
Also I think we cannot use choices for YamlFileArgument's because we transfrom the actual argument value by loading yaml file.